src/Runner.Worker/StepsRunner.cs

```diff
@@ -195,6 +195,26 @@ public async Task RunAsync(IExecutionContext jobContext)
                         }
                         else
                         {
+                            // This is our last, best chance to expand the display name.  (At this point, all the requirements for successful expansion should be met.)
```

Comment:
As per @AvaStancu's recommendation:
> Would it be possible to extract your implementation to a separate method and add some unit tests for it?

I agree with you @jww3 on your comments below regarding the difficulty and usefulness of unit testing here 
What do you think about a `TrySetDisplayname()` function?